vhdl_syntax/src/parser/productions/scalar_types.rs

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c)  2024, Lukas Scheller lukasscheller@icloud.com
+/// Parsing of scalar types
+use crate::parser::Parser;
+use crate::syntax::node_kind::NodeKind::*;
+use crate::tokens::{Keyword as Kw, TokenKind::*, TokenStream};
+
+impl<T: TokenStream> Parser<T> {
+    pub fn range(&mut self, max_length: usize) {
```

Comment:
I prefer this to be some internal function (i.e., `range_max_range` or so) and a secondary `range` function that passes `usize::MAX` to the internal function.

Reply:
Can do that. Not sure how useful that will be, but at least it would be consistent.

Reply:
You can also omit the public functions for now, but these functions that are not "standard conform" I would make `pub(crate)` for now. Making them public later is always easy. The inverse is not true.